### PR TITLE
Create install-64 for Zoom

### PR DESCRIPTION
**Pull request by Jai-JAP**

### CODE REVIEW

File: apps/Zoom/install-64 [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+function error {
+  echo -e "\\e[91m$1\\e[39m"
+  exit 1
+}
+
+cd $HOME
+
+# Get dependencies
+cd $HOME
+"${DIRECTORY}/pkg-install" "cmake:armhf pulseaudio-utils:armhf pulseaudio:armhf libxcb-shape0:armhf libxcb-randr0:armhf libxcb-image:armhf libxcb-image0:armhf libxcb-xtest0:armhf libxcb-keysyms1:armhf libxcb-xfixes0:armhf libturbojpeg0:armhf libdbus-1-3:armhf" "$(dirname "$0")" || exit 1
```
[the remark this answers is not in the document]
Comment (CleanMachine1) [reply apps/Zoom/install-64:14]: !

[the remark this answers is not in the document]
Comment (Jai-JAP) [reply apps/Zoom/install-64:14]: I too felt the same thing but it is extracted from [Zoom install-32](https://github.com/Botspot/pi-apps/blob/master/apps/Zoom/install-32#L14) 

```
"${DIRECTORY}/pkg-install" "libxcb-xtest0 libxcb-xfixes0 libturbojpeg0 cmake pulseaudio-utils pulseaudio" "$(dirname "$0")" || exit 1
```

**Comment by Jai-JAP** [comment]
> Have you tested this?

No, but it seems to be good to install perfectly.

I will test it & let you know the results.

File: apps/Zoom/install-64 (hunk 1 above)
[the remark this answers is not in the document]
Comment (Itai-Nelken) [reply apps/Zoom/install-64:14]: So can you remove it from both scripts please?

[the remark this answers is not in the document]
Comment (Jai-JAP) [reply apps/Zoom/install-64:14]: Sure

**Comment by Jai-JAP** [comment]
I also made some additional changes in install-32 as libturbojpeg has different names in Debian (Raspbian) and Ubuntu
It is libturbojpeg [in Debian (Raspbian)](https://packages.debian.org/search?keywords=libturbojpeg) and [in Ubuntu](https://packages.ubuntu.com/search?keywords=libturbojpeg)

**Comment by Jai-JAP** [comment]
@Itai-Nelken I removed cmake from both install scripts but this results in errors and zoom does not launch.
So restoring cmake in both install scripts.

**Comment by Jai-JAP** [comment]
Now at this stage Zoom is installed successfully and runs perfectly fine.

File: apps/Zoom/install-32 [hunk 2]
```diff
@@ -10,8 +10,14 @@ function error {
...
 # Get dependencies
-cd $HOME
-"${DIRECTORY}/pkg-install" "libxcb-xtest0 libxcb-xfixes0 libturbojpeg0 cmake pulseaudio-utils pulseaudio" "$(dirname "$0")" || exit 1
+DISTRO=$(lsb_release -is)
+if [[ "${DISTRO}" == "Debian" || "${DISTRO}" == "Raspbian" ]]; then
+  "${DIRECTORY}/pkg-install" "cmake libxcb-xtest0 libxcb-xfixes0 libturbojpeg0 pulseaudio-utils pulseaudio" "$(dirname "$0")" || exit 1
+elif [[ "${DISTRO}" == "Ubuntu" ]]; then
+  "${DIRECTORY}/pkg-install" "cmake libxcb-xtest0 libxcb-xfixes0 libturbojpeg pulseaudio-utils pulseaudio" "$(dirname "$0")" || exit 1
+else 
+	error "This script can't run on your OS! It HAS to be Debian or Ubuntu"
```
Comment (Itai-Nelken) [inline apps/Zoom/install-32:19]:
```suggestion
	error "This script can't run on your OS! It HAS to be Debian and derivatives like RPiOS or Ubuntu"
```

File: apps/Zoom/install-64 [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+cd $HOME
+
+# Get dependencies
+DISTRO=$(lsb_release -is)
+if [[ "${DISTRO}" == "Debian" || "${DISTRO}" == "Raspbian" ]]; then
+  "${DIRECTORY}/pkg-install" "cmake:armhf pulseaudio-utils:armhf pulseaudio:armhf libxcb-shape0:armhf libxcb-randr0:armhf libxcb-image0:armhf libxcb-xtest0:armhf libxcb-keysyms1:armhf libxcb-xfixes0:armhf libturbojpeg0:armhf libdbus-1-3:armhf" "$(dirname "$0")" || exit 1
+elif [[ "${DISTRO}" == "Ubuntu" ]]; then
+  "${DIRECTORY}/pkg-install" "cmake:armhf pulseaudio-utils:armhf pulseaudio:armhf libxcb-shape0:armhf libxcb-randr0:armhf libxcb-image0:armhf libxcb-xtest0:armhf libxcb-keysyms1:armhf libxcb-xfixes0:armhf libturbojpeg:armhf libdbus-1-3:armhf" "$(dirname "$0")" || exit 1
+else 
+	error "This script can't run on your OS! It HAS to be Debian or Ubuntu"
```
Comment (Itai-Nelken) [inline apps/Zoom/install-64:19]:
```suggestion
	error "This script can't run on your OS! It HAS to be Debian and derivatives like RPiOS or Ubuntu"
```

**Review by Itai-Nelken — APPROVED** [review]
Great! looks good to me, just a small change to the error given if the distro is not supported by the script.

File: apps/Zoom/install-64 (hunk 3 above)
Comment (Jai-JAP) [reply apps/Zoom/install-64]: Done

File: apps/Zoom/install-32 (hunk 2 above)
Comment (Jai-JAP) [reply apps/Zoom/install-32]: Done

**Comment by Jai-JAP** [comment]
I changed dependency pulseaudio:armhf pulseaudio-utils:armhf to arm64 ones to improve support for all apps as armhf version conflicts with arm64 version

File: apps/Zoom/install-32 (hunk 2 above)
Comment (Itai-Nelken) [reply apps/Zoom/install-32]: Thanks!

**Comment by Itai-Nelken** [comment]
can you test one last time please @Jai-JAP?

**Comment by Jai-JAP** [comment]
Tested! Works well.

**Comment by Itai-Nelken** [comment]
Thanks @Jai-JAP!

**Comment by Botspot** [comment]
RPiOS 64-bit running in vdesktop. I tried installing Zoom, but it failed with this error.
```
Installing Zoom with install-64 script
Running pkg-install...
Waiting until APT locks are released... Done
Running sudo apt update...
Some packages can be upgraded. Please consider running sudo apt full-upgrade.

Creating dummy deb...
dpkg-deb: building package 'pi-apps-c08187a2' in '/home/pi/pi-apps-c08187a2.deb'.

Installing dummy deb...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 pi-apps-c08187a2 : Depends: cmake:armhf
E: Unable to correct problems, you have held broken packages.

Apt finished.
Failed to install the packages!
APT reported these errors:
E: Unable to correct problems, you have held broken packages.

Failed to install Zoom with install-64 script!

Close this window to exit.
```

**Comment by Jai-JAP** [comment]
Can you please send your sources.list file

**Comment by Itai-Nelken** [comment]
@Botspot

**Comment by Jai-JAP** [comment]
> RPiOS 64-bit running in vdesktop. I tried installing Zoom, but it failed with this error.
> 
> ```
> Installing Zoom with install-64 script
> Running pkg-install...
> Waiting until APT locks are released... Done
> Running sudo apt update...
> Some packages can be upgraded. Please consider running sudo apt full-upgrade.
> 
> Creating dummy deb...
> dpkg-deb: building package 'pi-apps-c08187a2' in '/home/pi/pi-apps-c08187a2.deb'.
> 
> Installing dummy deb...
> Some packages could not be installed. This may mean that you have
> requested an impossible situation or if you are using the unstable
> distribution that some required packages have not yet been created
> or been moved out of Incoming.
> The following information may help to resolve the situation:
> 
> The following packages have unmet dependencies:
>  pi-apps-c08187a2 : Depends: cmake:armhf
> E: Unable to correct problems, you have held broken packages.
> 
> Apt finished.
> Failed to install the packages!
> APT reported these errors:
> E: Unable to correct problems, you have held broken packages.
> 
> Failed to install Zoom with install-64 script!
> 
> Close this window to exit.
> ```
> 
This issue is now solved. ( #724 )

**Comment by Itai-Nelken** [comment]
Oh, i forgot

**Pull request merged**